closed/make/Docs.gmk

```diff
@@ -20,4 +20,77 @@
 # 
 # ===========================================================================
 
-JVMTI_HTML := $(TOPDIR)/closed/variant-server/gensrc/jvmtifiles/jvmti.html
\ No newline at end of file
+JVMTI_HTML := $(TOPDIR)/closed/variant-server/gensrc/jvmtifiles/jvmti.html
+
+################################################################################
+# Text for openj9 modules
+
+OPENJ9_BASE_URL := https://www.eclipse.org/openj9/
+OPENJ9_BUG_SUBMIT_URL := https://github.com/eclipse/openj9/issues
+OPENJ9_FULL_COMPANY_NAME := IBM Corp
+OPENJ9_JAVADOC_BOTTOM := \
+    <a href="$(OPENJ9_BASE_URL)">Link</a> to Eclipse OpenJ9 documentation.<br> \
+    To raise a bug report or suggest an improvement create an <a href="$(OPENJ9_BUG_SUBMIT_URL)">Eclipse Openj9 issue.</a><br> \
+    Copyright &copy; 1998, $(COPYRIGHT_YEAR), $(OPENJ9_FULL_COMPANY_NAME). All rights reserved.
+
+################################################################################
+# The javadoc tool only allows for a single setting for headers and footers
+# per invocation. The sed command below replaces the footer for the pages
+# relating to openj9 classes to the correct value for those classes.
+#
+# '.' need escaping in the sed regex. If more special characters are introduced
+# in the openjdk JAVADOC_BOTTOM in future these would need escaping also.
+#
+# '&' needs escaping in sed replacement text.
+#
+# {@docroot} is expanded by javadoc when the html files are created, so the matching
+# needs to cater for anything {@docroot} may have been substituted by - '../../../..'
+# or '../..' or '..' etc.
+
+# \(\.\./\)*\.\. means zero  or more '../' strings followed by '..'
+
+JAVADOC_DOCROOT=\(\.\./\)*\.\.
+JAVADOC_BOTTOM_REGEX = $(subst {@docroot},$(JAVADOC_DOCROOT),$(subst .,\.,$(JAVADOC_BOTTOM)))
+OPENJ9_JAVADOC_BOTTOM := $(subst &,\&,$(OPENJ9_JAVADOC_BOTTOM))
+OPENJ9_SUBST_COMMAND = sed -i 's|$(JAVADOC_BOTTOM_REGEX)|$(OPENJ9_JAVADOC_BOTTOM)|g'
```

Comment:
I have to wonder whether the pattern we need to replace is instead `(\.\./)*` because references in the generated html will be to files with a sequence of zero or more `../` at the beginning of those paths?

Again, please use `:=` instead of `=` in macro definitions (unless inappropriate).

Reply:
I've changed the JAVADOC_DOCROOT assignment to `JAVADOC_DOCROOT := \(\.\./\)*\.\.`.
`=` is required for the JAVADOC_BOTTOM_REGEX assignment (and therefore the OPENJ9_SUBST_COMMAND assignment) because JAVADOC_BOTTOM is not defined in make/Docs.gmk until after closed/make/Docs.gmk has been included.

Reply:
The javadoc tool replaces @{docroot} with `..`, `../..`, `../../..` type references.  The regex `\(\.\./\)*` does not match unless `@{docroot}/` is replaced with the regex rather than just `@{docroot}`.  Since the intent is for the regex to replace the original `@{docroot}` then it seems best to leave it as it is.